File: frontend/lib/features/learning/domain/models/lesson.dart
class Lesson {
  final String? id;
  final String batchId;
  final String title;
  final String? description;
  final String? videoUrl;
  final int orderSequence;

  Lesson({
    this.id,
    required this.batchId,
    required this.title,
    this.description,
    this.videoUrl,
    required this.orderSequence,
  });

  factory Lesson.fromJson(Map<String, dynamic> json) {
    return Lesson(
      id: json['id'],
      batchId: json['batchId'] ?? '',
      title: json['title'] ?? '',
      description: json['description'],
      videoUrl: json['videoUrl'],
      orderSequence: json['orderSequence'] ?? 0,
    );
  }
}

File: backend/src/ShriNrityalaya.Domain/Entities/Lesson.cs
using ShriNrityalaya.Domain.Common;

namespace ShriNrityalaya.Domain.Entities;

public class Lesson : AuditableEntity
{
    public Guid? BatchId { get; set; }
    public Batch? Batch { get; set; }

    public string Title { get; set; } = string.Empty;
    public string? Description { get; set; }
    public string? VideoUrl { get; set; }
    
    public int OrderSequence { get; set; }
    
    public ICollection<LearningResource> Resources { get; set; } = new List<LearningResource>();
}
